core_aiml/word_sub.py

```diff
@@ -57,41 +57,22 @@ def __init__(self, defaults={}):
 
     def __call__(self, match):
         """Handler invoked for each regex match."""
-        return self[match.group(0)]
+        ret = self[match.group(0)]
+        return ret
 
     def __setitem__(self, i, y):
         self._regexIsDirty = True
         # for each entry the user adds, we actually add three entries:
         super(type(self), self).__setitem__(i.lower(), y.lower())  # key = value
         super(type(self), self).__setitem__(string.capwords(i), string.capwords(y))  # Key = Value
-        super(type(self), self).__setitem__(i.upper(), y.upper())  # KEY = VALUE
+
+        # When changing the person, "I" should never translate to an all caps words in the other person
+        if i != "I":
+            super(type(self), self).__setitem__(i.upper(), y.upper())  # KEY = VALUE

```

Comment:
Tests caught it. Also found that "be" is not conjugated when changing person, so it will translate "I am tired" to "You am tired" when changing person.